lib/pgBackRest/Common/Ini.pm

```diff
@@ -297,306 +391,249 @@ sub iniSave
     }
 
-    # Set default for required
-    $bRequired = defined($bRequired) ? $bRequired : true;
+    if (defined($strSubKey) && !defined($strKey))
+    {
+        confess &log(ASSERT, "strKey is required when strSubKey '${strSubKey}' is requested");
+    }
 
-    # Store the result
-    my $oResult = undef;
+    # Get the result
+    my $oResult = $self->{oContent}->{$strSection};
 
-    if (defined($strSubValue))
+    if (defined($strKey) && defined($oResult))
     {
-        if (!defined($strValue))
-        {
-            confess &log(ASSERT, "subvalue '${strSubValue}' requested but value is not defined");
-        }
+        $oResult = $oResult->{$strKey};
 
-        if (defined(${$oContent}{$strSection}{$strValue}))
+        if (defined($strSubKey) && defined($oResult))
         {
-            $oResult = ${$oContent}{$strSection}{$strValue}{$strSubValue};
+            $oResult = $oResult->{$strSubKey};
         }
     }
-    elsif (defined($strValue))
+
+    # When result is not defined
+    if (!defined($oResult))
     {
-        if (defined(${$oContent}{$strSection}))
+        # Error if a result is required
+        if (!defined($bRequired) || $bRequired)
         {
-            $oResult = ${$oContent}{$strSection}{$strValue};
+            confess &log(ASSERT, "strSection '$strSection'" . (defined($strKey) ? ", strKey '$strKey'" : '') .
+                                  (defined($strSubKey) ? ", strSubKey '$strSubKey'" : '') . ' is required but not defined');
         }
-    }
-    else
-    {
-        $oResult = ${$oContent}{$strSection};
-    }
-
-    if (!defined($oResult) && $bRequired)
-    {
-        confess &log(ASSERT, "manifest section '$strSection'" . (defined($strValue) ? ", value '$strValue'" : '') .
-                              (defined($strSubValue) ? ", subvalue '$strSubValue'" : '') . ' is required but not defined');
-    }
 
-    if (!defined($oResult) && defined($oDefault))
-    {
-        $oResult = $oDefault;
+        # Return default if specified
+        if (defined($oDefault))
+        {
+            return $oDefault;
+        }
     }
 
     return $oResult
 }
 
 ####################################################################################################################################
-# boolGet
-#
-# Get a numeric value.
+# boolGet() - get a boolean value.
 ####################################################################################################################################
-sub boolGet
-{
-    my $self = shift;
-    my $strSection = shift;
-    my $strValue = shift;
-    my $strSubValue = shift;
-    my $bRequired = shift;
-    my $bDefault = shift;
-
-    return $self->get($strSection, $strValue, $strSubValue, $bRequired,
-                      defined($bDefault) ? ($bDefault ? INI_TRUE : INI_FALSE) : undef) ? true : false;
-}
+sub boolGet {
+    return shift->get(shift, shift, shift, shift, defined($_[0]) ? (shift() ? INI_TRUE : INI_FALSE) : undef) ? true : false}
```

Comment:
Fair enough - I wasn't total convinced but you comment in the unit tests did convince me.  These functions need to be self-documenting.